doc/development/release_distribution.rst

```diff
@@ -15,137 +15,149 @@ Preamble
-It is ok to have two-digit identifiers; 4.6.10 is the next number after 4.6.9.
-The file should contain only the version number in this format, with no extra characters (the automatic line-break at the end is fine).
+Now jump to release_ if you are making a major or minor release, or bugfix_ if you are only fixing bugs in a previous release.
 
 .. _release:
 
 Create a new minor or major release
 -----------------------------------
 
-Create a new branch on the ``qutip/qutip`` repository using GitHub, e.g. 'qutip-4.1.X', beginning at the commit you want to use as the base of the release.
-This will likely be something fairly recent on the ``master`` branch.
-See `the GitHub help pages <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository#creating-a-branch>`_ for more information.
-Checkout the branch and push to your fork ::
-
-    $ git fetch upstream
-    $ git checkout -b qutip-4.1.X upstream/qutip-4.1.X
-    $ git push -u origin qutip-4.1.X
-
-Create a new branch from this, e.g. ::
-
-    $ git checkout -b 4.1-release_ready qutip-4.1.X
+This involves making a new branch to hold the release and adding some commits to set the code into "release" mode.
+This release should be done by branching directly off the ``master`` branch at its current head.
 
-First change the ``VERSION`` file to contain the new version number, and remove any ``dev`` postfix if present.
-A major release increments the first number, while a minor release increments the second.
-All numbers after the change digit are reset to 0, so the next minor release after 4.5.3 is 4.6.0, and the next major release after either of these is 5.0.0.
-Alpha, beta and release-candidate releases have a postfix ``.a<n>``, ``.b<n>`` or ``.rc<n>`` repsectively, where ``<n>`` is a counter for the pre-release status, starting from 0.
-For example, the third beta release of version 5.2.1 would have a version of ``5.2.1.b2``.
-The file should contain only the version number in this format, with no extra characters (the automatic line-break at the end is fine).
+#. On your machine, make sure your copy of ``master`` is up-to-date (``git checkout master; git pull upstream master``).
+   This should at least involve fetching the changelog PR that you just made.
+   Now create a new branch off a commit in ``master`` that has the state of the code you want to release.
+   The command is ``git checkout -b qutip-<major>.<minor>.X``, for example ``qutip-4.7.X``.
+   This branch name will be public, and must follow this format.
+#. Push the new branch (with no commits in it relative to ``master``) to the main ``qutip/qutip`` repository (``git push upstream qutip-4.7.X``).
+   This is the only situation in which it is ok to push to ``qutip/qutip`` without making a pull request; you cannot make a pull request until the branch has been created.
```

Comment:
```suggestion
```
I would just remove this sentence. It's more to read and I think it's clear that releasing is a special process anyway. I'm also fairly sure that this is not the *only* case that it is okay to do this (e.g. `dev.major` presumably was created this way too).

Reply:
I'm not sure, to be honest.  `dev.major` was a hugely special case, whereas doing a release should be around every two months, including patch releases. I think `dev.major` was the first time QuTiP ever had two principle branches for simultaneous development (as opposed to short-lived features), and we don't really anticipate ever needing to make a third branch like it, since the branch just represents the "next major" release.

I would like to leave in some allusion to the fact that the guide tells you to push a branch directly, and this shouldn't be standard procedure.

Reply:
Happy with this -- don't feel super strongly about the point.

Reply:
I'll just change it to "one of the few" and delete everything after the semicolon - makes it shorter, but still draws attention to it.